### ISSUE TITLE
[FR] dynamic sidebar width 

### ISSUE THREAD

**Issue by Offerel** [FreshRSS/FreshRSS#2426]
As a low priority feature, i would like to ask, if it is possible to make the wide of the sidebar dynamic, so that i can adjust the wide with the mouse. The wide of the sidebar should be saved as a personal option somewhere, so that it get applied, when i login on other computer as well.

**Comment by Alkarex** [comment]
Hello,
While waiting for a potential option, changing the width of the sidebar is something that can be achieved with a custom CSS rule using the custom-css extension.

**Comment by Offerel** [comment]
Maybe https://jqueryui.com/resizable/ is what we are looking for. I haven't used it in my projects yet. But maybe it's easier than I thought.

**Comment by aledeg** [comment]
Probably not the way we want to go since we want to remove completely jquery.

**Comment by Offerel** [comment]
Ohhh, I didn't know that. Only discovered the project for me a few days ago. Well, then we have to find something else.

**Comment by Frenzie** [comment]
You can probably do it in a few lines with some kind of drag area on the right of the element (like the border) that responds to mousedown + mousemove.

Storing that would be a lot more effort, but I can probably write a sample JS for use with the CustomJS extension pretty quickly if you're interested.

**Comment by Offerel** [comment]
The sample JS would be really nice.

**Comment by Frenzie** [comment]
I'll do it tomorrow.

**Comment by Frenzie** [comment]
@Offerel Here's a basic proof of concept. You'd need to expand it into a full extension to store the setting on the server.

```js
// did a quick search before starting, so as to write it in 10 mins instead of 30
// based on https://stackoverflow.com/a/53220241
const aside_feed = document.querySelector('#aside_feed');
const nav_entries = document.querySelector('#nav_entries');
let aside_feed_width = localStorage.getItem('xx_customjs_aside_feed_width');
const border_width = 10; // for easy to grab proof of concept

aside_feed.style.width = aside_feed_width + 'px';
nav_entries.style.width = aside_feed.style.width; // shouldn't that just be 100% in the CSS?

let mouse_x;
function resize(ev) {
	const dx = mouse_x - ev.x;
	mouse_x = ev.x;
	let new_width = parseInt(getComputedStyle(aside_feed, '').width) - dx;
	aside_feed.style.width = new_width + 'px';
	nav_entries.style.width = aside_feed.style.width; // shouldn't that just be 100% in the CSS?
	// store for next time
	localStorage.setItem('xx_customjs_aside_feed_width', new_width);
}

aside_feed.addEventListener('mousedown', function(ev) {
	if (ev.offsetX < aside_feed.offsetWidth + border_width) {
		mouse_x = ev.x;
		document.addEventListener('mousemove', resize);
	}
});

document.addEventListener('mouseup', function() {
	document.removeEventListener('mousemove', resize);
});
```

**Comment by Offerel** [comment]
Thats works great. With customCSS i have tryed to add a sort of "grabber", but i cant get to stick to the right side of #aside_feed. I can stick it to the left side of #stream with #stream::before, but then i cant resize the sidebar with it. I'm no that CSS king to stick it at the right side of #aside_feed.

```
#aside_feed::after {
	width: 5px;
	background: #000;
	content: "";
	height: 100%;
	position: absolute;
	cursor: w-resize;
}
```

**Comment by Frenzie** [comment]
Which browser? Depending on the specifics it might be necessary to use a real element (inserted through the DOM) instead of a CSS pseudo-element.

Maybe I'll have another look later, but it's going to be pretty hot… ;-)

**Comment by Offerel** [comment]
Mostly i use Firefox or some of its forks. But it should work also with Chromium and other browsers.

**Comment by Offerel** [comment]
Ok, i think, i have it now. I use both extensions CustomCSS and CustomJS.

CustomJS:
```
const aside_feed = document.querySelector('#aside_feed');
const nav_entries = document.querySelector('#nav_entries');
const subcriptions = document.querySelector('#btn-subscription');
const border_width = 10; // for easy to grab proof of concept

var grabber = document.createElement("span");
grabber.className = "asidegrabber";

if(aside_feed !== null) {
	aside_feed.appendChild(grabber);
	aside_feed.addEventListener('mousedown', function(ev) {
		if (ev.offsetX < aside_feed.offsetWidth + border_width) {
			mouse_x = ev.x;
			document.addEventListener('mousemove', resize);
		}
	});
}

if(nav_entries !== null) {
	nav_entries.style.width = aside_feed.style.width;
}

let mouse_x;
function resize(ev) {
	const dx = mouse_x - ev.x;
	mouse_x = ev.x;
	let new_width = parseInt(getComputedStyle(aside_feed, '').width) - dx;
	aside_feed.style.width = new_width + 'px';
	grabber.style.left = new_width - 5 + 'px';
	subcriptions.style.width = new_width - 55 + 'px';
	if(typeof nav_entries !== null && nav_entries !== 'undefined')
		nav_entries.style.width = aside_feed.style.width;
}

document.addEventListener('mouseup', function() {
	document.removeEventListener('mousemove', resize);
});
```

CustomCSS:
```
a#btn-subscription {width: 176px;}
a#btn-importExport {width: 15px;}
.asidegrabber {
	width: 2px;
	background: transparent;
	content: "";
	height: 50px;
	position: fixed;
	cursor: w-resize;
	left: 225px;
	border-left: 2px solid #365571;
	border-right: 2px solid #19222a;
	top: 50%;
	transform: translateY(-50%);
}
```

**Comment by Offerel** [comment]
We can close this with #4463
